### ISSUE TITLE
missing config entry causes config.getExpression() to return None, resulting in non-intuitive errors

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/MPAS-Analysis#226]
If a config option is missing but config.getExpression() is used to try to parse that option, the result is `None`, whereas the user would probably expect an exception to be raised about the missing section or option.

The result is a very non-intuitive error later in the code when an attempt is made to use the result of the expression, for example:
```
  File "/home/xylar/code/mpas-work/analysis/add_mpas_climatology_task/mpas_analysis/ocean/climatology_map.py", line 100, in create_tasks
    if len(comparisonGridNames) == 0:
TypeError: object of type 'NoneType' has no len()
```
It seems like we should remove the if/else case here:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/develop/mpas_analysis/configuration/MpasAnalysisConfigParser.py#L93
and let an exception be raised if the section/option doesn't exist.

**Comment by xylar** [comment]
@pwolfram, could you remind me why we decided to have getExpression() return None if the option didn't exist?  I grepped the code and didn't see anywhere (including testing) where this seemed to be needed.

**Comment by pwolfram** [comment]
@xylar, thanks for asking this question.  It is needed to handle chunking: https://github.com/MPAS-Dev/MPAS-Analysis/blob/b78aa6b6963e94a6c974de2dc096b52448a58230/mpas_analysis/shared/generalized_reader/generalized_reader.py#L193 and https://github.com/MPAS-Dev/MPAS-Analysis/blob/6a9a591e94a18b42e69c08473bfd3e9b92010bcf/mpas_analysis/shared/mpas_xarray/mpas_xarray.py#L579 but this can be refractorred.  I think if you change this you'll end up with a case where you get an error if chunking isn't specified.  Perhaps the easiest way around this would be a try/execpt block at https://github.com/MPAS-Dev/MPAS-Analysis/blob/b78aa6b6963e94a6c974de2dc096b52448a58230/mpas_analysis/shared/generalized_reader/generalized_reader.py#L191 but we may be ok if https://github.com/MPAS-Dev/MPAS-Analysis/blob/b78aa6b6963e94a6c974de2dc096b52448a58230/mpas_analysis/shared/generalized_reader/generalized_reader.py#L189 is set properly.  

Philosophically, I agree that a missing option shouldn't just report `None`.   What do you think about issuing a warning instead?  This would have made catching the error above easier but still provide the conceptual clarity that if you ask for an option in the list that doesn't exist you get `None` as the outcome.  I fully agree that we need at least a warning to be issue in this case, but I'm not 100% convinced issuing a full exception is really what is best.  Is the strongest argument for issuing the error that it requires more robust coding as well as user engagement with respect to choice of namelist parameters?  I think you loose a little flexibility but perhaps eliminate some failure modes this way and it seems like it all boils down to high-level philosophy across the software in terms of its design.

**Comment by xylar** [comment]
Thanks @pwolfram, that clears up why `getExpression` has the behavior it currently does.

Presently, we use `getExpression` related to chunking in only 2 spots, both in the MOC task:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/b78aa6b6963e94a6c974de2dc096b52448a58230/mpas_analysis/ocean/meridional_overturning_circulation.py#L251
and
https://github.com/MPAS-Dev/MPAS-Analysis/blob/b78aa6b6963e94a6c974de2dc096b52448a58230/mpas_analysis/ocean/meridional_overturning_circulation.py#L251
whereas we use `getExpression` for other things in many places.  It seems to me that the simplest solution would be to replace these two instances with:
```python
if config.has_option(sectionName, 'maxChunkSize):
    chunking = config.getExpression(sectionName, 'maxChunkSize')
else:
    chunking = None
```
and to have `getExpression` raise an error (as normal) if the option doesn't exist.  Would that solution work for you?

**Comment by pwolfram** [comment]
It sounds like an overall much cleaner solution cross-repo-- I'd say go for it!  Thanks for following up with me so that we don't have weird side effects with chunking- it is much appreciated!

**Comment by xylar** [comment]
Closed via #227
